tests/components/homekit/test_homekit.py

```diff
@@ -526,3 +756,132 @@ def _mock_get_accessory(*args, **kwargs):
             "linked_battery_sensor": "sensor.light_battery",
         },
     )
+
+
+async def test_setup_imported(hass):
+    """Test async_setup with imported config options."""
+    legacy_persist_file_path = hass.config.path(HOMEKIT_FILE)
+    legacy_aid_storage_path = hass.config.path(STORAGE_DIR, "homekit.aids")
+    legacy_homekit_state_contents = {"homekit.state": 1}
+    legacy_homekit_aids_contents = {"homekit.aids": 1}
+    await hass.async_add_executor_job(
+        _write_data, legacy_persist_file_path, legacy_homekit_state_contents
+    )
+    await hass.async_add_executor_job(
+        _write_data, legacy_aid_storage_path, legacy_homekit_aids_contents
+    )
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        source=SOURCE_IMPORT,
+        data={CONF_NAME: BRIDGE_NAME, CONF_PORT: DEFAULT_PORT, CONF_ENTRY_INDEX: 0},
+        options={},
+    )
+    entry.add_to_hass(hass)
+
+    with patch(f"{PATH_HOMEKIT}.HomeKit") as mock_homekit:
+        mock_homekit.return_value = homekit = Mock()
+        type(homekit).async_start = AsyncMock()
+        assert await hass.config_entries.async_setup(entry.entry_id)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5453535)